nx-integration-tests-setup/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Nx Integration Tests Setup"
+author: "Mage-OS"
+description: "Setup and cache Nx project, this can then be reused with all service combinations."
+
+inputs:
+  repository:
+    type: string 
+    description: "Repository"
+    required: true
+  ref:
+    type: string
+    description: "head SHA"
+    required: true
+  pr_base:
+    type: string
+    description: "pr base SHA"
+    required: true
+
+# permissions:
+#   contents: write
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout PR commit
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ inputs.repository }}
+        ref: ${{ inputs.ref }}
+        path: main
+        # Need to do this otherwise Nx cant determine diff
+        fetch-depth: 0
+
+    - name: Fetch base
+      working-directory: ./main
+      shell: bash
+      run: git fetch origin ${{ inputs.pr_base }}
+
+    - name: Cache Composer dependencies
+      uses: actions/cache@v3
+      with:
+        path: /tmp/composer-cache
+        key: ${{ runner.os }}-${{ hashFiles('**/composer.lock') }}
+
+    - id: get-composer-and-php-version
+      name: Get Composer & PHP Version
+      working-directory: ./main
+      shell: bash
+      run: |
+        echo "php_version=$(jq -c .services.php[0] supported-services.json)" >> "$GITHUB_OUTPUT"
+        echo "composer_version=$(jq -rc .services.composer[0] supported-services.json)" >> "$GITHUB_OUTPUT"
+    - name: Composer Install
+      uses: php-actions/composer@v6
+      with:
+        version: ${{ steps.get-composer-and-php-version.outputs.composer_version }}
+        php_version: ${{ steps.get-composer-and-php-version.outputs.php_version }}
+        args: "--ignore-platform-reqs --optimize-autoloader"
+        working_dir: main
+
+    # could probably swap this to a docker container in future
+    - name: Install NX
+      working-directory: ./main
+      shell: bash
+      run: |
+        npm install -g nx@15.4.1
+        
+    # should be able to cache this in future also
+    - name: Checkout Nx Repo
+      uses: actions/checkout@v3
+      with:
+        repository: adamzero1/nx-for-php

```

Comment:
Should this stay there or is this something we should also have as part of mage-os?

Reply:
This could be moved, to Mage-OS.
No issues with that, just hadn't pushed as I believed @damienwebdev was implementing in a better way, which would make this redundant.

Reply:
Okay, cool. Absolutely fine for me, was just a thought.